### DISCUSSION TITLE
API Document Management

### DISCUSSION THREAD

**Discussion by BAEKJungHo** [withDogecoin/chaterview#11] [Ideas]
# Solutions

1. __Asciidoctor + Spring REST Docs__

- https://asciidoctor.org/
- https://spring.io/guides/gs/testing-restdocs/

2. __OpenAPISpec__

- https://swagger.io/specification/
- [OpenApiSpec.을 이용한 더욱 효과적인 API 문서화 with Kotlin](https://github.com/traeper/api_documentation)

3. __OpenAPISpec + RestDocs + MkDocs__

- https://www.mkdocs.org/
- https://github.com/mkdocs/mkdocs/wiki/MkDocs-Themes
- [Kakaopay - OpenAPI Specification을 이용한 더욱 효과적인 API 문서화](https://tech.kakaopay.com/post/openapi-documentation/)

4. __Orchid - How to Document a Kotlin Project__

- https://orchid.run/wiki/learn/tutorials/how-to-document-kotlin

**Comment by BAEKJungHo** [comment]
I want to do number three.

Apply OpenAPISpec on the application server and use Mkdocs, which allows you to collect and view various teams and documents with a separate document server.

It also supports a lot of great themes.

**Reply by mangchhe** [reply]
@BAEKJungHo 
I've looked at the links you gave me and I'd also like number 3. 👍👍

**Reply by BAEKJungHo** [reply]
@mangchhe Awesome Docs [Click](https://withdogecoin.github.io/)
